src/Symfony/Component/Console/Application.php

```diff
@@ -975,34 +990,47 @@ protected function doRunCommand(Command $command, InputInterface $input, OutputI
 
         $commandSignals = $command instanceof SignalableCommandInterface ? $command->getSubscribedSignals() : [];
         if ($commandSignals || $this->dispatcher && $this->signalsToDispatchEvent) {
-            if (!$this->signalRegistry) {
-                throw new RuntimeException('Unable to subscribe to signal events. Make sure that the "pcntl" extension is installed and that "pcntl_*" functions are not disabled by your php.ini\'s "disable_functions" directive.');
-            }
+            $signalRegistry = $this->getSignalRegistry();
```

Comment:
`getSignalRegistry` throws a nearly identical exception, so it made sense to me to have it centralized.